lib/bot-service.js

```
@@ -26,7 +26,16 @@ botService.get = async (resource) => {
             });
             response.on('end', () => {
                 const data = decodeToken(body);
-                data === undefined ? reject() : resolve(data);
+                var responseBody = {};
+                if (data instanceof Error)  {
```

does this ever happen?

---

Yes, this happens in either one of these 2 situations - Missing key `bot` in response JSON or Unable to decode response from BoT server.

---

I think the decodeToken function should throw that error then, and not here. Now we have it in two places.

---

Changed in decodeToken to throw error and removed checking for error within get/post